### PR TITLE
[release/v2.21] fix for duplicate entries in front-loadbalancer

### PR DESCRIPTION
**Pull request by sachintiptur**
**What this PR does / why we need it**:
Fixes duplicate entries for SourceRanges in front-loadbalancer service. This is manual back porting of  [11308](https://github.com/kubermatic/kubermatic/pull/11308)
**Which issue(s) this PR fixes**:

Fixes #[11307](https://github.com/kubermatic/kubermatic/issues/11307)

**What type of PR is this?**
/kind bug

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Fix duplicate SourceRange entries for front-loadbalancer Service
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/resources/nodeportproxy/nodeportproxy.go [hunk 1]
```diff
@@ -415,9 +415,11 @@ func FrontLoadBalancerServiceCreator(data *resources.TemplateData) reconciling.N
 				}
 			}
 
+			// set of Source IP ranges
+			sourceIPList := sets.String{}
 			if data.Seed().Spec.NodeportProxy.Envoy.LoadBalancerService.SourceRanges != nil {
 				for _, cidr := range data.Seed().Spec.NodeportProxy.Envoy.LoadBalancerService.SourceRanges {
-					s.Spec.LoadBalancerSourceRanges = append(s.Spec.LoadBalancerSourceRanges, string(cidr))
+					sourceIPList.Insert(string(cidr))
```
Comment (embik) [inline pkg/resources/nodeportproxy/nodeportproxy.go:422]: This misses being written to `s.Spec.LoadBalancerSourceRanges`, I believe.

Comment (embik) [reply pkg/resources/nodeportproxy/nodeportproxy.go:422]: In addition, is this the only place where `s.Spec.LoadBalancerSourceRanges` can be set?

Comment (sachintiptur) [reply pkg/resources/nodeportproxy/nodeportproxy.go:422]: fixed

**Review by embik — APPROVED** [review]
/approve

**Pull request merged**